File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/wrapper/util/FoodStats.java
package rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.util;


import rbq.wtf.lycoris.client.wrapper.MapEnum;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapField;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapMethod;
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.WrapperClass;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.IWrapper;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.util.event.HoverEvent;
import rbq.wtf.lycoris.client.wrapper.wrappers.wrapper.util.event.click.ClickEvent;

import java.lang.reflect.Field;
import java.lang.reflect.Method;

@WrapperClass(mcpName = "net.minecraft.util.FoodStats", targetMap = MapEnum.VANILLA189)
public class FoodStats extends IWrapper {
    @WrapField(mcpName = "foodLevel", targetMap = MapEnum.VANILLA189)
    public static Field foodLevel;

    public FoodStats(Object obj) {
        super(obj);
    }

    public int getFoodLevel() {
        return (int) getField(foodLevel);
    }
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/util/FoodStats.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.util

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.annotation.WrapField
import rbq.wtf.lycoris.client.wrapper.annotation.WrapperClass
import rbq.wtf.lycoris.client.wrapper.IWrapper
import java.lang.reflect.Field

@WrapperClass(mcpName = "net.minecraft.util.FoodStats", targetMap = MapEnum.VANILLA189)
class FoodStats(obj: Any) : IWrapper(obj) {
    companion object {
        @WrapField(mcpName = "foodLevel", targetMap = MapEnum.VANILLA189)
        lateinit var foodLevel: Field
    }

    val foodLevel: Int
        get() = getField(Companion.foodLevel) as Int
}